### PR TITLE
Deploy to Goerli

### PR DESCRIPTION
**Pull request by scott-silver**
attempting to create a Goerli/USDC deployment, in part because we'll need it to test a Mumbai (Avalanche testnet) l2 deployment and in part because there's a chance that Etherscan may become unusable for Kovan soon.

The set of [ChainLink feeds for Goerli](https://docs.chain.link/docs/ethereum-addresses/#Goerli%20Testnet) is currently very limited, and doesn't include COMP 😞:

image

~~So this deployment does not include COMP (or any other reward token).~~ [now with COMP!] It uses LINK as the first collateral asset (the first collateral asset is weirdly important for scenarios, since we reference it via `asset0` frequently).

### CODE REVIEW

File: scenario/BulkerScenario.ts [hunk 1]
```diff
@@ -38,7 +38,7 @@ scenario(
     expect(await comet.balanceOf(albert.address)).to.be.equal(0n);
 
     // Albert's actions:
-    // 1. Supplies 100 units of collateral
+    // 1. Supplies 1000 units of collateral
```
Comment (scott-silver) [inline scenario/BulkerScenario.ts:41]: This test currently assumes that 100 units of `asset0` are worth 1500 of the base unit.

The price of LINK on Goerli is currently around $7.

Bumping up the supplied collateral to cover the borrow,

File: scenario/SupplyScenario.ts [hunk 2]
```diff
@@ -295,15 +295,14 @@ scenario(
...
       betty.supplyAssetFrom({
         src: albert.address,
         dst: betty.address,
         asset: collateralAsset.address,
         amount: 100n * scale,
-      }),
-      [/ERC20: transfer amount exceeds allowance/, /transfer amount exceeds spender allowance/]
-    );
+      })
+    ).to.be.reverted;
```
Comment (scott-silver) [inline scenario/SupplyScenario.ts:305]: These scenarios are great, but they assume that the token returns an error message of some sort. LINK reverts without a reason string for these two instances.

Comment (jflatow) [reply scenario/SupplyScenario.ts:305]: Oy

**Comment by kevincheng96** [comment]
> * do we want to have COMP as an asset, even if there is currently no ChainLink feed for COMP? (if so, should we use the ChainLink feed of a different asset?)
>   
>   * other option, from Adam: use SimplePriceFeed until ChainLink publishes a COMP feed

I think using a `SimplePriceFeed` until we have Chainlink support is best. It'll be easy to switch the price feed once Chainlink supports COMP.

> * related: do we want to have a reward token on this deployment? Is that reward asset COMP, or something with a ChainLink price feed?

Yeah, it'll be nice to be able to test rewards on testnet. If we set up a `SimplePriceFeed` for COMP, we should be able to use COMP as the reward asset without any issues. Might be interesting/cool to test a different reward asset on testnet, but it probably doesn't make sense on Goerli since it should mirror mainnet.

File: .github/workflows/deploy-market.yaml [hunk 3]
```diff
@@ -9,6 +9,7 @@ on:
           - fuji
           - kovan
           - mainnet
+          - goerli
```
Comment (kevincheng96) [inline .github/workflows/deploy-market.yaml:12]: You might to also add an entry for `goerli` in line 33 (`ethereum_url` field). Same for the other workflows.

File: README.md [hunk 4]
```diff
@@ -50,10 +50,6 @@ Currently, Avalanche mainnet and testnet (fuji) are supported. This means that d
 
 To use this project with other chains, the block explorer API key for your target chain must be set in .env (e.g. `SNOWTRACE_KEY` for Avalanche).
 
-An example deployment command looks like:
-
-`yarn deploy --network fuji`
```
Comment (kevincheng96) [inline README.md:55]: Is this no longer true?

File: deployments/goerli/usdc/configuration.json [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    "borrowKink": 0.8,
+    "borrowSlopeLow": 0.035,
+    "borrowSlopeHigh": 0.25,
+    "borrowBase": 0.015
+  },
+  "tracking": {
+    "indexScale": "1e15",
+    "baseSupplySpeed": "0.000011574074074074073e15",
+    "baseBorrowSpeed": "0.0011458333333333333e15",
+    "baseMinForRewards": "1000000e6"
```
Comment (kevincheng96) [inline deployments/goerli/usdc/configuration.json:23]: Should we lower this on testnets to $10k or $100k instead?

File: deployments/goerli/usdc/deploy.ts [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+    }
+  );
+
+  const WBTC = await deploymentManager.clone('WBTC', clone.wbtc, []);
+  const WETH = await deploymentManager.clone('WETH', clone.weth, []);
+  const LINK = await deploymentManager.clone('LINK', clone.link, []);
+
+  // Deploy all Comet-related contracts
+  const deployed = await deployComet(deploymentManager, deploySpec);
+  // const { comet, rewards } = deployed;
```
Comment (kevincheng96) [inline deployments/goerli/usdc/deploy.ts:58]: Can delete?

**Review by kevincheng96 — COMMENTED** [review]
🎉

File: README.md (hunk 4 above)
Comment (scott-silver) [reply README.md:55]: yeah, `yarn deploy` isn't a command anymore

Comment (jflatow) [reply README.md:55]: `yarn hardhat deploy`

Comment (scott-silver) [reply README.md:55]: so really this line should read:

```
yarn hardhat deploy --network fuji --deployment usdc
```

File: .github/workflows/deploy-market.yaml (hunk 3 above)
Comment (scott-silver) [reply .github/workflows/deploy-market.yaml:12]: good catch.

that line is going to be awful when we've got 5-6 deployments (with their corresponding testnets)

Comment (scott-silver) [reply .github/workflows/deploy-market.yaml:12]: updated

File: README.md (hunk 4 above)
Comment (scott-silver) [reply README.md:55]: updated

File: deployments/goerli/usdc/configuration.json (hunk 5 above)
Comment (scott-silver) [reply deployments/goerli/usdc/configuration.json]: yep, that works. updated to $10K

File: deployments/goerli/usdc/deploy.ts (hunk 6 above)
Comment (scott-silver) [reply deployments/goerli/usdc/deploy.ts]: deleted

**Comment by scott-silver** [comment]
Added COMP as a collateral asset; updated to use SimplePriceFeed as its feed

**Comment by jflatow** [comment]
Superseded by #565

**Pull request closed without merging**